module/VuFind/src/VuFind/View/Helper/Root/Citation.php

```diff
@@ -192,6 +192,47 @@ public function getCitation($format)
         return '';
     }
 
+    protected function trimPunctuation($text)
+    {
+        return trim($text, " \n\r\t\v\0,:;/");
+    }
+
+    protected function removeDateRange($name)
+    {
+        return preg_replace('/\d+[ ]*\-[ ]*\d*/', '', $name);
+    }
+
+    protected function nameToGivenFamily($name)
```

Comment:
I suspect this is going to need some more logic added. What if the name is already in "First Last" format? What if there is a suffix like ", Jr." or a title like ", Mrs."?

Reply:
I am looking at the other functions in this class, and they don't seem to cover all those cases. Doesn't mean we shouldn't though! I'll see if I can find a more robust parser.

Reply:
I think a lot of the other functions are already assuming names are formatted in a particular way, and most of the name processing is designed to fix non-conformant strings. But if we need to parse out specific elements, it can get complicated.

Reply:
I found a composer library: https://github.com/joshfraser/PHP-Name-Parser.

Reply:
Yeah, the lack of "Last, First" support is disappointing, since we'd have to write our own parser to deal with complex names with multiple components (e.g. "Miller, Alex. McVeigh, Mrs." or "Smith, John, III") before we could use the third-party parser.

Reply:
This one will likely work better for our purposes: https://github.com/theiconic/name-parser. I've found a few more robust libraries in other languages, but I don't know if it's worth the conversion effort. I see a lot of unit tests in my future.

Reply:
That library does look potentially like a better fit, though the lack of activity in two years is somewhat concerning.